octobot_trading/exchanges/implementations/future_ccxt_exchange.py

```diff
@@ -13,178 +13,43 @@
-
-    def get_exchange_pair(self, pair) -> str:
-        return self.connector.get_exchange_pair(pair)
-
-    def get_pair_cryptocurrency(self, pair) -> str:
-        return self.connector.get_pair_cryptocurrency(pair)
-
-    def get_default_balance(self):
-        return self.connector.get_default_balance()
-
-    def get_rate_limit(self):
-        return self.connector.get_rate_limit()
-
-    async def switch_to_account(self, account_type: trading_enums.AccountTypes):
-        return await self.connector.switch_to_account(account_type=account_type)
-
-    def parse_balance(self, balance):
-        return personal_data.parse_decimal_portfolio(self.connector.parse_balance(balance))
-
-    def parse_trade(self, trade):
-        return self.connector.parse_trade(trade)
-
-    def parse_order(self, order):
-        return self.connector.parse_order(order)
-
-    def parse_ticker(self, ticker):
-        return self.connector.parse_ticker(ticker)
-
-    def parse_ohlcv(self, ohlcv):
-        return self.connector.parse_ohlcv(ohlcv)
-
-    def parse_order_book(self, order_book):
-        return self.connector.parse_order_book(order_book)
-
-    def parse_order_book_ticker(self, order_book_ticker):
-        return self.connector.parse_order_book_ticker(order_book_ticker)
-
-    def parse_timestamp(self, data_dict, timestamp_key, default_value=None, ms=False):
-        return self.connector.parse_timestamp(data_dict, timestamp_key, default_value=default_value, ms=ms)
-
-    def parse_currency(self, currency):
-        return self.connector.parse_currency(currency)
-
-    def parse_order_id(self, order):
-        return self.connector.parse_order_id(order)
-
-    def parse_order_symbol(self, order):
-        return self.connector.parse_order_symbol(order)
-
-    def parse_status(self, status):
-        return self.connector.parse_status(status)
-
-    def parse_side(self, side):
-        return self.connector.parse_side(side)
-
-    def parse_account(self, account):
-        return self.connector.parse_account(account)
-
-    def clean_recent_trade(self, recent_trade):
-        return self.connector.clean_recent_trade(recent_trade)
-
-    def clean_trade(self, trade):
-        return self.connector.clean_trade(trade)
-
-    def clean_order(self, order):
-        return self.connector.clean_order(order)
-
     async def get_positions(self, **kwargs: dict) -> list:
         return await self.connector.get_positions(**kwargs)
 
     async def get_symbol_positions(self, symbol: str, **kwargs: dict) -> list:
         return await self.connector.get_symbol_positions(symbol=symbol, **kwargs)
+    
+    async def get_positions_linear_settle_coins(self) -> None or list:
+        """
+        override if exchange has different settle coins for linear positions
+
+        Returns:
+            None or list: list of currencies
+        """
+        # todo find a way to automate
+        return ["USDT", "USDC"]
```

Comment:
?

Reply:
on the latest ccxt with bybit get_positions it requires to provide the settle currency, not sure if other exchanges require it as well.

As only bybit is supported anyway, we can keep the default for now I'd say.
I did overwrite it in the bybit tentacle as well, so if we change the default at one point, we are good

And for later it would be nice to find a way to detect settled coins automatically

Reply:
Ok I see. As you said, I think we can keep the default or raise a not implemented error while we don't have a wait to detect or fetch the settled coins automatically. And yes I think it has to be overwritten in the exchange tentacle

Reply:
i moved it to the connector settings, now only generic settings have coins in there